### ISSUE TITLE
FetchXml for Liquid as an one of the outputs

### ISSUE THREAD

**Issue by ZaarourOmar** [rappen/FetchXMLBuilder#264]
Hi,
It would be great if one of the output is a liquid formatted FetxhXml call that can be copied and pasted in a portal web template and ready for consumption.

**Comment by rappen** [comment]
Sounds like a good idea!
I have not worked with portals that much - how would that format look?

**Comment by ZaarourOmar** [comment]
> Sounds like a good idea!
> I have not worked with portals that much - how would that format look?

Yes, so basically, one of the very common uses in Portal is to create a webtemplate that contains a fetxhxml mixed with liquid syntax to be used to query data from CRM and use it on the page. People with less knowledge in liquid will benefit from having a fetxhxml query converted to a liquid fetxhxml query with results being formatted as JSON or XML. This post by Colin Vermander outlines how it is formatted and used.

https://colinvermander.com/2017/04/17/dynamics-365-portals-use-liquid-to-return-json-or-xml/

**Comment by rappen** [comment]
To me it looks like plain fetchXML there, with some embedded liquid parametrization. But how would FXB know where the developer would want to inject liquid?
Perhaps @koolin could have some ideas or clarify what I am not understanding?

**Comment by koolin** [comment]
Perhaps the ask is to sub/replace values of attributes with liquid syntax attributes? Or to create the wrapping required to add the fetch in liquid?

`{% fetchxml my_query %}
  <fetch version="1.0" mapping="logical">
    
  </fetch>
{% endfetchxml %}
  
{{ my_query.xml | escape }}
{{ my_query.results.total_record_count }}
{{ my_query.results.more_records }}
{{ my_query.results.paging_cookie | escape }}
{% for result in my_query.results.entities %}
  {{ result.id | escape }}
{% endfor %}`

`<condition attribute="dpx_courselocationid" operator="eq" value="{{ request.params['location'] | xml_escape }}" />`

For the condition really sure something like that actually would be feasible considering the potential variations and unknowns in how someone might use the liquid or the logic they require.  I use fetchXML builder today to build the normal fetch and then go about adding specific logic as required in liquid.

**Comment by ZaarourOmar** [comment]
I actually meant the wrapping and the JSON/XML creation.  I agree that the above mentioned snippet by Colin is not always the case but it is easier to modify that snippet than to build it from scratch. Thanks Colin and Jonas :)

**Comment by rappen** [comment]
Looking at this issue - only 5 years later ;) - and I know that Power Pages are moving more and more from Liquid to Power Fx.
Would it be good to have some variation to work even better with Power Fx?
I think I'll just close this issue since Liquid is old stuff... 😏

Pinging @ZaarourOmar @koolin @nickdoelman for your thoughts!
